### ISSUE TITLE
Server panic in 0.9.0 on `show series`, causes rollback with data lost

### ISSUE THREAD

**Issue by ckmaresca** [influxdb/influxdb#2983]
I just installed 0.9.0 and was running some tests on it to see how it behaved.   I am using the Go InfluxDB client (rc33, didn't see a tag for 0.9.0...) .  I should note I'm a novice Go dev, so my client code is less than perfect, but it basically tries to insert a whole bunch of events extracted from a JSON file.  These events have timestamps, so I am setting the `Time` in the insert as well.

I just inserted a bunch of entries which were showing up in the web interface, but when I tried to query them, Influx crashed and apparently rolled back the insert.

Here is some of the influxdb.log:
```
[http] 2015/06/14 20:35:00 127.0.0.1 - root [14/Jun/2015:20:35:00 -0700] POST /write?consistency=&db=test&precision=&rp=default HTTP/1.1 500 55 - InfluxDBClient 8097b3af-130f-11e5-801c-000000000000 2.860268ms
[http] 2015/06/14 20:36:52 10.10.2.98 - - [14/Jun/2015:20:36:52 -0700] GET /query?u=root&p=root&q=show%20series&db=test HTTP/1.1 200 211 http://10.10.2.222:8083/ Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/40.0.2214.91 Safari/537.36 c309eace-130f-11e5-801d-000000000000 1.613299ms
panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x1 pc=0x70adc1]

goroutine 332 [running]:
github.com/boltdb/bolt.(*Tx).Rollback(0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/boltdb/bolt/tx.go:210 +0x31
github.com/influxdb/influxdb/tsdb.(*LocalMapper).Close(0xc20812cc60)
```
There are a bunch more stack traces all related to this `Rollback` (I think).  The startup looks like this:
```
created by github.com/influxdb/influxdb/tsdb.(*QueryExecutor).ExecuteQuery
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/query_executor.go:211 +0x22b
GOMAXPROCS set to 2
2015/06/14 20:41:52 [INFO] raft: Node at 10.10.2.222:8088 [Follower] entering Follower state
2015/06/14 20:41:52 read local node id: 1
2015/06/14 20:41:53 [WARN] raft: Heartbeat timeout reached, starting election
2015/06/14 20:41:53 [INFO] raft: Node at 10.10.2.222:8088 [Candidate] entering Candidate state
2015/06/14 20:41:53 [DEBUG] raft: Votes needed: 1
2015/06/14 20:41:53 [DEBUG] raft: Vote granted. Tally: 1
2015/06/14 20:41:53 [INFO] raft: Election won. Tally: 1
2015/06/14 20:41:53 [INFO] raft: Node at 10.10.2.222:8088 [Leader] entering Leader state
2015/06/14 20:41:53 [INFO] raft: Disabling EnableSingleNode (bootstrap)
2015/06/14 20:41:53 [DEBUG] raft: Node 10.10.2.222:8088 updated peer set (2): [10.10.2.222:8088]
2015/06/14 20:41:53 [DEBUG] raft: Node 10.10.2.222:8088 updated peer set (2): [10.10.2.222:8088]
[httpd] 2015/06/14 20:41:56 listening on HTTP: [::]:8086
```
Not sure what is causing the issue, and I'm not exactly sure how much to put in this issue, so please ask if you need more detail.

**Comment by ckmaresca** [comment]
Looking around, this may be the same as #2971, not sure as I didn't test the command line.

**Comment by toddboom** [comment]
@ckmaresca I think this is fixed in v0.9.0 - I couldn't reproduce it. If you can, see if you can give us some more steps to recreate.
